packages/core/src/rules/common/no-required-schema-properties-undefined.ts

```diff
@@ -0,0 +1,30 @@
+import { Oas3Rule } from '../../visitors';
+import { Oas3Schema, Oas3_1Schema } from '../../typings/openapi';
+import { Oas2Schema } from 'core/src/typings/swagger';
+import { UserContext } from 'core/src/walk';
+
+export const NoRequiredSchemaPropertiesUndefined: Oas3Rule = () => {
+  return {
+    Schema: {
+      enter(schema: Oas3Schema | Oas3_1Schema | Oas2Schema, { location, report }: UserContext) {
+        if (schema?.required) {
+          const missingRequiredProperties: string[] = schema.required.filter((property) => {
+            return !schema.properties || schema.properties[property] === undefined;
+          });
+
+          if (missingRequiredProperties.length) {
+            const reportMessage =
+              missingRequiredProperties.length > 1
+                ? `Required properties are undefined: ${missingRequiredProperties.join(', ')}.`
+                : `Required property ${missingRequiredProperties.join()} is undefined.`;
+
+            report({
+              message: reportMessage,
+              location: location.child('required'),
+            });
+          }
+        }
```

Comment:
Could you check if this will produce a better output? 

```suggestion
         schema.required?.forEach((requiredProperty, i) => {
          if (!schema.properties || schema.properties[requiredProperty] === undefined) {
            report({
              message: `Required property '${requiredProperty}' is undefined.`,
              location: location.child(['required', i]),
            });
          }
        })
```

Reply:
Fixed, thanks!